components/relay/src/lib.rs

```diff
@@ -13,12 +13,33 @@ use serde::{Deserialize, Serialize};
 use url::Url;
 use viaduct::{header_names, Method, Request};
 
+/// Represents a client for the Relay API.
+///
+/// Use this struct to connect and authenticate with a Relay server,
+/// managing authorization to call protected endpoints.
+///
+/// # Authorization
+/// - Clients should user the `FirefoxAccount.getAccessToken()` function
+///   to obtain a relay-scoped access token (scope: `https://identity.mozilla.com/apps/relay`).
+/// - Then, construct the [`RelayClient`] with the access token.
+///   All requests will then be authenticated to the Relay server via `Authorization: Bearer {fxa-access-token}`.
+/// - The Relay server verifies this token with the FxA OAuth `/verify` endpoint.
+/// - Clients are responsible for getting a new access token when needed.
 #[derive(uniffi::Object)]
 pub struct RelayClient {
+    /// Base URL for the Relay server.
     server_url: String,
+    /// Optional authentication token for API requests.
     auth_token: Option<String>,
 }
 
+/// Represents a Relay email address object returned by the Relay API.
+///
+/// Includes metadata and statistics for an alias, such as its status,
+/// usage stats, and identifying information.
+///
+/// See:
+/// https://dev.fxprivaterelay.nonprod.cloudops.mozgcp.net/api/v1/docs/redoc/#tag/emails/operation/relayaddresses_retrieve
```

Comment:
should this link to a .md file or similar in some git repo? This url smells fragile.